src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -97,6 +99,36 @@ private static MethodInfo CreateGetWebHookDataMethodInfo()
 
             arguments.Add(triggerParameter.Name, triggerArgument);
 
+            //add query string parameters that aren't satisfied by any bindings to avoid having WebJobs SDK throw an 
+            //error for any parameters it doesn't find values from through bindings.
+            var bindingNames = function.Metadata.Bindings.Select(b => b.Name).ToImmutableSortedSet();
+
+            //extract all possible parameter values from the route
+            var route = RoutingUtility.ExtractRouteTemplateFromMetadata(function.Metadata);
+            var parameters = RoutingUtility.ExtractRouteParameters(route, request);          
+            foreach (var pair in parameters)
+            {
+                arguments.Add(pair.Key, pair.Value);
+            }
+
+            //Extract parameters from the query string if match a parameter name that is not a binding
+            var queryStringParameters = request.GetQueryNameValuePairs().ToImmutableDictionary();
+            foreach (var parameter in function.Parameters)
+            {
+                // if the parameter is not a binding or already found in the routes then assume it is a parameter to be extracted from the query string
+                if (!bindingNames.Contains(parameter.Name) && !arguments.ContainsKey(parameter.Name))
```

Comment:
Have you tested this with arguments that expect their value from named inputs (e.g. properties from an input binding, system named inputs, etc.)? For a few reasons (including safety/security) we want to make sure those take priority.